server/gitrest/packages/gitrest-base/src/routes/summaries.ts

```diff
@@ -277,20 +289,29 @@ export function create(
             return;
         }
         const wholeSummaryPayload: IWholeSummaryPayload = request.body;
-        const resultP = getRepoManagerFromWriteAPI(repoManagerFactory, repoManagerParams, repoPerDocEnabled)
-            .then(async (repoManager): Promise<IWriteSummaryResponse | IWholeFlatSummary> => {
-                const fsManager = fileSystemManagerFactory.create(repoManagerParams.fileSystemManagerParams);
+        const resultP = (async () => {
+            // There are possible optimizations we can make throughout the summary write process
+            // if we are using repoPerDoc model and it is the first summary for that document.
+            const optimizeForInitialSummary = isInitialSummary && repoPerDocEnabled;
+            // If creating a repo per document, we do not need to check for an existing repo on initial summary write.
+            const repoManager = await getRepoManagerFromWriteAPI(repoManagerFactory, repoManagerParams, repoPerDocEnabled, optimizeForInitialSummary);
+            const fsManager = fileSystemManagerFactory.create(repoManagerParams.fileSystemManagerParams);
+            // A new document cannot already be soft-deleted.
+            if (!optimizeForInitialSummary) {
```

Comment:
Will this not apply even if we are not using repo per doc? If that is the case, shouldn't this be !isInitialSummary?

Reply:
Fair point. Technically, checkSoftDeleted already only works if `repoPerDoc==true`, so I figured this was still OK. You're right though, that if soft-delete support were added for repoPerTenant, then this would break it.

Reply:
I would personally like to keep this change contained to the AFR use-case, where repoPerDoc is enabled out of necessity.